### ISSUE TITLE
Feature request: Provisioning of FILE FORMAT Snowflake object

### ISSUE THREAD

**Issue by doctoryes** [Snowflake-Labs/terraform-provider-snowflake#116]
The functionality to add a Snowpipe as the `CREATE FILE FORMAT` SQL performs is currently missing from this provider. It'd be nice to be able to manage these objects via Terraform.

https://docs.snowflake.net/manuals/sql-reference/sql/create-file-format.html

**Comment by sjauld** [comment]
I'll pick this up once my last PR is merged, should be pretty straight forward.

**Comment by ryanking** [comment]
@sjauld we will get that other PR reviewed soon

**Comment by nelg** [comment]
We also would like to use this feature.

**Comment by robbruce** [comment]
> I'll pick this up once my last PR is merged, should be pretty straight forward.

@sjauld are you making much progress on this?

**Comment by sjauld** [comment]
I have done some work on it, but it's on an old laptop, I'll have a look tonight. It was a while ago but I think it was pretty close to done maybe.

**Comment by shaunmaris** [comment]
This would be a great feature to include!

**Comment by brucedvgw** [comment]
Will this go in the next release? It's currently the missing piece for us when we do the S3 integration. 👍

**Comment by ryanking** [comment]
@brucedvgw it will when someone writes the code, are you volunteering? :)

**Comment by brucedvgw** [comment]
Haha fair call @ryanking 😁. I would love to contribute to this project if I had the skill set to do so, although it's a great learn opportunity! Thanks for your work on the provider so far 🙏

**Comment by robbruce** [comment]
@ryanking I'm thinking of picking this up as we have this coded already on our own provider, so would be a case of porting over. However from a design decision point of view, could create resources as either:

`snowflake_file_format` - that supports lots of config (there is a lot when you take in all the sub options that are relevant to each format type)

or
`snowflake_file_format_csv`, `snowflake_file_format_json`, `snowflake_file_format_avro` etc.

Thoughts?

**Comment by spaztic1215** [comment]
@robbruce the flexibility and extensibility of `snowflake_file_format` seems like the right approach to me.  Just checking back in if you're still game to porting over the FILE FORMAT resource, otherwise we might take this on since it's bit us a couple times not having formats managed by this provider.  CC @ajbosco

**Comment by ryanking** [comment]
@robbruce I think a single resource makes sense. At least that seems to be the common approach when building terraform providers.

**Comment by JavierLopezT** [comment]
Great feature. Would love to have this (not able to contribute though, I have no knowledge)

**Comment by ajbosco** [comment]
Attempted to add this resource here: https://github.com/chanzuckerberg/terraform-provider-snowflake/pull/392

**Comment by philippeboyd** [comment]
Seems to be available in latest release
PR merged : https://github.com/chanzuckerberg/terraform-provider-snowflake/pull/577

**Comment by sfc-gh-swinkler** [comment]
closing this ticket. please open a new ticket if there are any further issues.
